packages/trackerless-network/test/integration/RandomGraphNode-Layer1Node-Latencies.test.ts

```diff
@@ -124,19 +124,20 @@ describe('RandomGraphNode-DhtNode-Latencies', () => {
             waitForCondition(() => node.getNumberOfOutgoingHandshakes() === 0)
         ))
 
-        await wait(20000)
-        let mismatchCounter = 0
-        graphNodes.forEach((node) => {
-            const nodeId = node.getOwnNodeId()
-            node.getTargetNeighborIds().forEach((neighborId) => {
-                if (neighborId !== entryPointRandomGraphNode.getOwnNodeId()) {
-                    const neighbor = graphNodes.find((n) => n.getOwnNodeId() === neighborId)
-                    if (!neighbor!.getTargetNeighborIds().includes(nodeId)) {
-                        mismatchCounter += 1
+        await waitForCondition(() => {
+            let mismatchCounter = 0
+            graphNodes.forEach((node) => {
+                const nodeId = node.getOwnNodeId()
+                node.getTargetNeighborIds().forEach((neighborId) => {
+                    if (neighborId !== entryPointRandomGraphNode.getOwnNodeId()) {
+                        const neighbor = graphNodes.find((n) => n.getOwnNodeId() === neighborId)
+                        if (!neighbor!.getTargetNeighborIds().includes(nodeId)) {
+                            mismatchCounter += 1
+                        }
                     }
-                }
+                })
             })
-        })
-        expect(mismatchCounter).toBeLessThanOrEqual(2)
+            return mismatchCounter <= 2
```

Comment:
Does this test allow that there are 2 neighbors, which shouldn't be neighbors? Or what exactly the `mismatchCounter` represents? Why that kind of mismatch is allowed?

Reply:
There's a known issue in low latency (simulator) environments where sometimes nodes do not seem to be able to form bideirectional connections. This is quite rare and debugging it in the past has been quite the rabbit hole

Reply:
Ok could add a link to that Liner ticket and TODO that we should change the test to not allow mismatch when the issue have been fixed